frontend/src/utils/index.ts

```diff
@@ -86,3 +86,45 @@ export const MBytesToBytes = megs => {
   // 1 Megabyte = 1048576 Bytes.
   return 1048576 * num
 }
+
+export function delay(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+export type WaitForFunction<T> = () => Promise<T>
+export type WaitForCheckFunction<T> = (x: T) => boolean
+export interface WaitForOptions<T> {
+  timeoutMs: number
+  intervalMs?: number
+  checkFn?: WaitForCheckFunction<T>
+}
+const defaultCheckFn: WaitForCheckFunction<unknown> = x => Boolean(x)
+
+/**
+ * Retry function until it returns a promise that resolves to a truthy value.
+ * Alternatively you can provide your own `checkFn` which should return `true`
+ * when you want to stop or `false` to continue retrying.
+ */
+export function waitFor<T>(
+  fn: WaitForFunction<T>,
+  { timeoutMs, intervalMs = 100, checkFn = defaultCheckFn }: WaitForOptions<T>,
```

Comment:
What's wrong with the pattern? 

The interval is somewhat arbitrary but I though 100ms is a good default. We are using `waitFor` when we think something might already be there and want to react relatively quickly when it does appear.

Reply:
It makes so many requests, sometimes the tab becomes unresponsive and I had to kill it.

Reply:
Hmm, I haven't seen that and 10 http requests per second doesn't seem like that many. While they are repeating they are also returning no content. 

I'll increase it and you can tell me if that solves it. I have a suspicion it might be something else that's causing your tab to hang though.

Reply:
I've increased it to a second. Let me know if you see the tab lock up again.